src/components/ReadingGoalPage/useReadingGoalReducer.ts

```diff
@@ -0,0 +1,176 @@
+/* eslint-disable max-lines */
+/* eslint-disable react-func/max-lines-per-function */
+import { Dispatch, SetStateAction, useReducer } from 'react';
+
+import BookIcon from '@/icons/book.svg';
+import ClockIcon from '@/icons/clock.svg';
+import SettingsIcon from '@/icons/settings-stroke.svg';
+import { ReadingGoalType } from '@/types/auth/ReadingGoal';
+
+interface ReadingGoalState {
+  period: 'daily' | 'continuous';
```

Comment:
we should move this to an enum. It will be painful if we need to rename any string without an enum